### PR TITLE
handle CantDeserializeException raised from deserialize method

### PR DESCRIPTION
**Pull request by sjhewitt**
re-implementation of #232 using @zzzeek's comments

### CODE REVIEW

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 4e8bc2aeb3bbd44520574ac5ede5f4e85ac346e0 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 4e8bc2aeb3bbd44520574ac5ede5f4e85ac346e0: https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

**Comment by sjhewitt** [comment]
just pushed an update to try to fix the linting errors - `tox -r -e pep8` runs fine locally now

**Review by sqla-tester — COMMENTED** [review]
Hi, this is **sqla-tester** and I see you've pinged me for review. However, user **sjhewitt** is not authorized to initiate CI jobs.  Please wait for a project member to do this!

**Comment by sqla-tester** [comment]
Patchset 555315d2db76ed111bb05967e48a88f02813f939 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

File: docs/build/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,16 @@ Changelog
     :version: 1.1.9
     :include_notes_from: unreleased
 
+    .. change::
+        :tags: feature, region
```
Comment (sqla-tester) [inline docs/build/changelog.rst:10]: **mike bayer** (zzzeek) wrote:

oh, technicality...this should be in a text file by itself inside of docs/build/unreleased/233.rst    it gets merged here using tooling later on.   

for an example see a previous commit like https://github.com/sqlalchemy/dogpile.cache/commit/f450a785aa9156d86216b194b9bf64791120abd9#diff-1568e46bf5b517b20c8f1ac9e9c2ff79134ac2be7695dee4c6cfa7bc5b8b92e0

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

File: tests/cache/test_memory_backend.py [hunk 2]
```diff
@@ -10,5 +17,30 @@ class MemoryBackendSerializerTest(_GenericSerializerTest, MemoryBackendTest):
...
+):
+    backend = "dogpile.cache.memory"
+
+    region_args = {
+        "serializer": lambda v: json.dumps(v).encode("ascii"),
+        "deserializer": raise_cant_deserialize_exception,
+    }
+
+    def test_serializer_cant_deserialize(self):
+        region = self._region()
```
Comment (sqla-tester) [inline tests/cache/test_memory_backend.py:35]: **mike bayer** (zzzeek) wrote:

this test is fine but feels like it should be part of the generic suite.

We could for example add this test to the existing _fixtures._GenericSerializerTest like this:

	diff --git a/tests/cache/_fixtures.py b/tests/cache/_fixtures.py
	index 9e71f8f..60ee281 100644
	--- a/tests/cache/_fixtures.py
	+++ b/tests/cache/_fixtures.py
	@@ -388,6 +388,18 @@ class _GenericSerializerTest(TestCase):
	         "deserializer": json.loads,
	     }
	 
	+    def test_serializer_cant_deserialize(self):
	+        region = self._region(region_args={
	+            "serializer": self.region_args["serializer"],
	+            "deserializer":raise_cant_deserialize_exception,
	+        })
	+
	+        value = {"foo": ["bar", 1, False, None]}
	+        region.set("k", value)
	+        asserted = region.get("k")
	+        eq_(asserted, NO_VALUE)
	+        
	+
	     def test_uses_serializer(self):
	         region = self._region()


that way it gets run for all the backends that include a _GenericSerializerTest class, which is all of them (memory, redis, memcached, dbm).

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

**Review by sqla-tester — COMMENTED** [review]
**mike bayer** (zzzeek) wrote:

cool thanks!  two changes to make.   then I think it will be ready

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

File: docs/build/changelog.rst (hunk 1 above)
Comment (sjhewitt) [reply docs/build/changelog.rst:10]: gotcha - will make the change

File: tests/cache/test_memory_backend.py (hunk 2 above)
Comment (sjhewitt) [reply tests/cache/test_memory_backend.py:35]: yeah, I didn't like it being in the memory serializer. good idea for the change. I'll implement it

**Comment by zzzeek** [comment]
waiting on your end here (no rush) - ping me when you update

**Comment by sjhewitt** [comment]
yup, got bogged down with other stuff. will aim to update the PR by the end of the week!

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

this has been in my queue for four months, if you want to pick it up again you can reopen and it should all be fine

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370 has been **abandoned**.  That means that at least for the moment I need to close this pull request. Sorry it didn't work out :(

**Pull request closed without merging**